tensorflow/python/keras/benchmarks/keras_examples_benchmarks/mnist_conv_custom_training_benchmark_test.py

```diff
@@ -0,0 +1,320 @@
+
+    Raise:
+      ValueError: if `dataset` is None or if `optimizer` instance is
+      not provided or if `loss_fn` instance is not provided.
+    """
+    if distribution_strategy is not None and \
+      not isinstance(dataset, tf.distribute.DistributedDataset):
+      raise ValueError('tf.distribute.DistributedDataset'
+                       ' required in distribution strategy.')
+
+    if distribution_strategy is None and \
+      not isinstance(dataset, tf.data.Dataset):
+      raise ValueError('`tf.data` is required.')
+
+    if not isinstance(loss_fn, tf.keras.losses.Loss):
+      raise ValueError('`tf.keras.losses.Loss` instance '
+                       'for loss_fn is required.')
+
+    if not isinstance(optimizer, tf.keras.optimizers.Optimizer):
+      raise ValueError('`tf.keras.optimizers` instance '
+                       'for optimizer is required.')
+
+    build_time_list, avg_epoch_time_list, train_step_time_list = [], [], []
+    wall_time_list, exp_per_sec_list, warmup_time_list = [], [], []
+
+    total_num_examples = epochs * self.num_examples
+
+    for _ in range(run_iters):
+      timer = timeit.default_timer
+      start_time = timer()
+
+      strategy_scope = distribution_util.get_strategy_scope(
+          distribution_strategy)
+      with strategy_scope:
+        t0 = timer()
+        model = model_fn()
+        build_time = timer() - t0
+      # Run one warm up epoch.
+      t1 = timer()
+      self.train_function(
+          model, dataset, loss_fn, optimizer,
+          1, distribution_strategy, batch_size)
+      warmup_time = timer() - t1
+
+      t2 = timer()
+      train_step_time = self.train_function(
+          model, dataset, loss_fn, optimizer,
+          epochs, distribution_strategy, batch_size)
+      end_time = timer()
+
+      train_step_time_list.append(train_step_time)
+      warmup_time_list.append(warmup_time)
+      build_time_list.append(build_time)
+      wall_time_list.append(end_time - start_time)
+      exp_per_sec_list.append(total_num_examples / (end_time - t2))
+      avg_epoch_time_list.append((end_time - t2) / epochs)
+
+    metrics = []
+    metrics.append({'name': 'build_time', 'value': np.mean(build_time_list)})
+    metrics.append({'name': 'avg_epoch_time',
+                    'value': np.mean(avg_epoch_time_list)})
+    metrics.append({'name': 'exp_per_sec', 'value': np.mean(exp_per_sec_list)})
+    metrics.append({'name': 'warmup_time', 'value': np.mean(warmup_time_list)})
+    metrics.append({'name': 'train_step_time',
+                    'value': np.mean(train_step_time_list)})
+    metrics.append({'name': 'epochs', 'value': epochs})
+
+    wall_time = np.mean(wall_time_list)
+
+    return metrics, wall_time
+
+  def benchmark_custom_training_mnist_bs_256(self):
+    """Measure performance with batch_size=128 and run_iters=1"""
+    batch_size = 256
+    run_iters = 1 # But we only support one iteration.
```

Comment:
why only 1 iter is supported? We've discussed this, and the fix is to build the model once but train it multiple iterations.

Reply:
Yeah, I changed the iterations my code locally and want to figure the performance issue before update this PR.